### PR TITLE
[chore] Revise task executor

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR revises task executor to conditionally provide last update.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/worker/task_executor.py [hunk 1]
```diff
@@ -193,7 +193,18 @@ async def execute(self) -> Any:
...
+            if (
+                task is not None
+                and task.progress is not None
+                and task.progress.get("percent") == 100
+            ):
+                # skip the last progress update if the progress is already 100%
+                skip_last_progress_update = True
+
+            if not skip_last_progress_update:
+                await self.task_progress_updater.update_progress(percent=100)
```
Comment (gee-senbong) [inline featurebyte/worker/task_executor.py:207]: Current sub-tasks rely on progress's metadata field to pass the information back to parent task. If the task's progress get overridden, the metadata value will be lost.

Comment (kchua78) [reply featurebyte/worker/task_executor.py:207]: Would it make sense to update the `update_progress` logic to retain existing message if the `message` parameter is not provided.

Comment (gee-senbong) [reply featurebyte/worker/task_executor.py:207]: Do you mean revise this fix to keep the existing metadata & message & only update percentage?

Comment (kchua78) [reply featurebyte/worker/task_executor.py:207]: Would it address the issue that you are trying to fix? Does it make sense for a more systematic fix to similar problems? If not we can keep the current approach

Comment (gee-senbong) [reply featurebyte/worker/task_executor.py:207]: The task will automatically update the percentage to 100% and override message & task metadata. If we want to change `update_progress` method, we need to do something like keep the last value if it is not provided. Just not sure whether there is any case where we want to set the last update message/metadata to be `None`.

Comment (kchua78) [reply featurebyte/worker/task_executor.py:207]: Indeed. That will require an additional parameter. Let's keep the current approach

**Review by kchua78 — APPROVED** [review]

**Pull request merged**